# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender_api/src/image.rs
# Commit: da24abf34123
# Full Hash: da24abf341236e00fa47e9e55dc72c82ae8a8c84
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-12-14 17:20:20
# Regressor Bug: 1866020
# File Overlap Count: 1
# Description:
#   Bug 1866020 - Override buggy colorspace conversion on Pixel devices. r=gw,padenot,geckoview-reviewers,owlish
#   
#   Pixel 6, 7, and 8 devices running Android 14 are affected by a bug
#   where video frames with SMPTE 432 color primaries are rendered
#   incorrectly when sampled from an external texture in GLES. To work
# ==============================================================================

diff -r 3a34e7a424e0 -r da24abf34123 gfx/wr/webrender_api/src/image.rs
--- a/gfx/wr/webrender_api/src/image.rs	Thu Dec 14 10:08:52 2023 +0000
+++ b/gfx/wr/webrender_api/src/image.rs	Thu Dec 14 10:45:08 2023 +0000
@@ -117,6 +117,10 @@
     /// understand, particularly YUV. See
     /// https://www.khronos.org/registry/OpenGL/extensions/OES/OES_EGL_image_external.txt
     TextureExternal = 2,
+    /// External texture which is forced to be converted from YUV to RGB using BT709 colorspace.
+    /// This maps to GL_TEXTURE_EXTERNAL_OES in OpenGL, using the EXT_YUV_TARGET extension.
+    /// https://registry.khronos.org/OpenGL/extensions/EXT/EXT_YUV_target.txt
+    TextureExternalBT709 = 3,
 }
 
 /// Storage format identifier for externally-managed images.